### ISSUE TITLE
Node server keeps crashing repeatedly with "Error: reserved fields must be empty"

### ISSUE THREAD

**Issue by booboothefool** [apollographql/subscriptions-transport-ws#94]
This started happening suddenly, and now I can't even get my server back up. It's been a few hours. It just restarts, then immediately crashes. Using 0.5.1 btw.

Any ideas?

![screen shot 2017-03-09 at 6 52 02 pm](https://cloud.githubusercontent.com/assets/13257645/23776040/b53ac3b0-04f9-11e7-8c47-d6bb5fcd42b8.png)

Very scary...
![screen shot 2017-03-09 at 8 26 49 pm](https://cloud.githubusercontent.com/assets/13257645/23778299/8892e1ea-0507-11e7-9da4-2da9d68c014a.png)

**Comment by booboothefool** [comment]
Where do I add this:

https://github.com/websockets/ws/issues/1039#issuecomment-285591423

For this library?

**Comment by booboothefool** [comment]
I'm trying to add the line here now: https://github.com/apollographql/subscriptions-transport-ws/blob/master/src/server.ts#L96

But installing my fork from GitHub doesn't install the /src directory... huh?

**Comment by booboothefool** [comment]
Alright, so after committing the entire folder to make sure the change was in, which btw was adding:

```
            request.on('error', function (err) {
                console.error(err);
            });
```
to https://github.com/apollographql/subscriptions-transport-ws/blob/master/src/server.ts#L96

the error is being caught, and my server stopped crashing. After about 10 or so 1 star reviews on my app within the past day, hahaha. 

I'm just happy the server doesn't crash, however would like to understand why this is happening if anyone knows. Thanks.

**Comment by Urigo** [comment]
@booboothefool that is a good idea.
Also looks like a pretty simple change, would you like to create a PR to expose that event on the subscriptionServer?

**Comment by Urigo** [comment]
Fixed in https://github.com/apollographql/subscriptions-transport-ws/pull/130

**Comment by dotansimha** [comment]
Thank you for opening this issue, @booboothefool . we fixed it in 0.7.0,
 and now we are handling the `error` callback of the socket,
 as you suggested. 
(https://github.com/apollographql/subscriptions-transport-ws/pull/130)
